sdk/src/main/java/io/opentelemetry/sdk/trace/export/BatchSpansProcessor.java

```diff
@@ -411,4 +321,184 @@ public void run() {
       }
     }
   }
+
+  @Immutable
+  public static final class Config {
+    private final long scheduleDelayMillis;
+    private final int maxQueueSize;
+    private final int maxExportBatchSize;
+    private final int exporterTimeoutMillis;
+    private final boolean sampled;
+
+    private Config(
+        boolean sampled,
+        long scheduleDelayMillis,
+        int maxQueueSize,
+        int maxExportBatchSize,
+        int exporterTimeoutMillis) {
+      this.sampled = sampled;
+      this.scheduleDelayMillis = scheduleDelayMillis;
+      this.maxQueueSize = maxQueueSize;
+      this.maxExportBatchSize = maxExportBatchSize;
+      this.exporterTimeoutMillis = exporterTimeoutMillis;
+    }
+
+    public long getScheduleDelayMillis() {
+      return scheduleDelayMillis;
+    }
+
+    public int getMaxQueueSize() {
+      return maxQueueSize;
+    }
+
+    public int getMaxExportBatchSize() {
+      return maxExportBatchSize;
+    }
+
+    public int getExporterTimeoutMillis() {
+      return exporterTimeoutMillis;
+    }
+
+    public boolean isReportOnlySampled() {
+      return sampled;
+    }
+
+    public static Config getDefault() {
+      return new Builder().build();
+    }
+
+    public static Builder newBuilder() {
+      return new Builder();
+    }
+
+    public static class Builder extends ConfigBuilder {
+
+      private static final String KEY_SCHEDULE_DELAY_MILLIS = "OTEL_BSP_SCHEDULE_DELAY";
+      private static final String KEY_MAX_QUEUE_SIZE = "OTEL_BSP_MAX_QUEUE";
+      private static final String KEY_MAX_EXPORT_BATCH_SIZE = "OTEL_BSP_MAX_EXPORT_BATCH";
+      private static final String KEY_EXPORT_TIMEOUT_MILLIS = "OTEL_BSP_EXPORT_TIMEOUT";
+      private static final String KEY_SAMPLED = "OTEL_BSP_REPORT_SAMPLED";
+
+      private static final long SCHEDULE_DELAY_MILLIS = 5000;
+      private static final int MAX_QUEUE_SIZE = 2048;
+      private static final int MAX_EXPORT_BATCH_SIZE = 512;
+      private static final int EXPORT_TIMEOUT_MILLIS = 30_000;
+      private static final boolean REPORT_ONLY_SAMPLED = true;
```

Comment:
It is hard to understand these names, I would propose to prefix all of them with "DEFAULT_"

Reply:
I'd also like to have them documented outside of these private fields. Maybe put them in a package-info file?